subprojects/core/src/main/java/org/gradle/execution/plan/Node.java

```diff
@@ -247,6 +247,11 @@ public void dependenciesProcessed() {
         return dependencySuccessors;
     }
 
+    @OverridingMethodsMustInvokeSuper
```

Comment:
We are defining a new term now. Let's add some Javadoc describing its relationship to the other terms.

Reply:
I added some Javadoc. Note that the term is not new, we already have `hasHardSuccessor()`.